src/actions/ActionKeys.ts

```diff
@@ -0,0 +1,62 @@
+export enum GlobalActionKeys {
+  INCREMENT_LOADING_COUNTER = 'INCREMENT_LOADING_COUNTER',
+  DECREMENT_LOADING_COUNTER = 'DECREMENT_LOADING_COUNTER'
+}
+
+export enum HelpDropdownActionKeys {
+  STATUS_REFRESH = 'STATUS_REFRESH'
+}
+
+export enum LoginActionKeys {
+  LOGIN_REQUEST = 'LOGIN_REQUEST',
+  LOGIN_SUCCESS = 'LOGIN_SUCCESS',
+  LOGIN_FAILURE = 'LOGIN_FAILURE',
+  LOGOUT_SUCCESS = 'LOGOUT_SUCCESS'
+}
+
+export const enum MessageCenterActionKeys {
+  ADD_MESSAGE = 'ADD_MESSAGE',
+  REMOVE_MESSAGE = 'REMOVE_MESSAGE',
+  MARK_MESSAGE_AS_READ = 'MARK_MESSAGE_AS_READ',
+  SHOW = 'SHOW',
+  HIDE = 'HIDE',
+  TOGGLE_EXPAND = 'TOGGLE_EXPAND',
+  TOGGLE_GROUP = 'TOGGLE_GROUP',
+  HIDE_NOTIFICATION = 'HIDE_NOTIFICATION'
+}
+
+export enum NamespaceActionKeys {
+  NAMESPACE_REQUEST_STARTED = 'NAMESPACE_REQUEST_STARTED',
+  NAMESPACE_SUCCESS = 'NAMESPACE_SUCCESS',
+  NAMESPACE_FAILED = 'NAMESPACE_FAILED'
+}
+
+export enum ServiceGraphActionKeys {
+  SERVICE_GRAPH_SIDE_PANEL_SHOW_INFO = 'SERVICE_GRAPH_SIDE_PANEL_SHOW_INFO'
+}
+
+export enum ServiceGraphDataActionKeys {
+  GET_GRAPH_DATA_START = 'GET_GRAPH_DATA_START',
+  GET_GRAPH_DATA_SUCCESS = 'GET_GRAPH_DATA_SUCCESS',
+  GET_GRAPH_DATA_FAILURE = 'GET_GRAPH_DATA_FAILURE',
+  HANDLE_LEGEND = 'HANDLE_LEGEND'
+}
+
+export enum ServiceGraphFilterActionKeys {
+  // Toggle Actions
+  TOGGLE_LEGEND = 'TOGGLE_LEGEND',
+  TOGGLE_GRAPH_NODE_LABEL = 'TOGGLE_GRAPH_NODE_LABEL',
+  TOGGLE_GRAPH_CIRCUIT_BREAKERS = 'TOGGLE_GRAPH_CIRCUIT_BREAKERS',
+  TOGGLE_GRAPH_VIRTUAL_SERVICES = 'TOGGLE_GRAPH_VIRTUAL_SERVICES',
+  TOGGLE_GRAPH_MISSING_SIDECARS = 'TOGGLE_GRAPH_MISSING_SIDECARS',
+  TOGGLE_TRAFFIC_ANIMATION = 'TOGGLE_TRAFFIC_ANIMATION',
+  SET_GRAPH_EDGE_LABEL_MODE = 'SET_GRAPH_EDGE_LABEL_MODE',
+  // Disable Actions
+  ENABLE_GRAPH_FILTERS = 'ENABLE_GRAPH_FILTERS',
+  // Refresh Rate
+  SET_GRAPH_REFRESH_RATE = 'SET_GRAPH_REFRESH_RATE'
+}
+
+export enum UserSettingsActionKeys {
+  NAV_COLLAPSE = 'NAV_COLLAPSE'
+}
```

Comment:
This looks OK to me, but alternatives are to use e.g. `GlobalActionsKeys.ts`, having them next to the actions or inside `action/keys`.
cc: @mtho11

Reply:
This refactor of ActionKeys is something totally separate and not required to fix this bug. 
I personally don't agree with this change. The actions keys were co-located with their actions which is the way I believe it should be -- no need to import extra files for the keys. We don't put all the reducers in a single file, we organize them by logical groups (not by kind). As the application grows I think this method of organization makes much more sense.

Reply:
I think is needed for removing the circular dependencies of the imports

Reply:
There is an import cycle that I didn't list in the main description:
<pre>
  utils/Authentication.ts ->
    store/ConfigStore.ts ->
    reducers/index.ts ->
    reducers/ServiceGraphDataState.ts ->
    actions/ServiceGraphDataActions.ts ->
    utils/Authentication.ts
</pre>

The file `reducers/ServiceGraphDataState.ts` depends on `actions/ServiceGraphDataActions.ts` but it only needs `ServiceGraphDataActionKeys` enum. And `actions/ServiceGraphDataActions.ts` has API calls that need the authentication token, thus, the dependency on `utils/Authentication.ts` making the import-cycle. This is the point I chose to break the cycle. By extracting `ServiceGraphDataActionKeys` the cycle is broken.

I extracted all other Action Keys only to make the code uniform. But if you don't like it, I can extract only what is needed.

Reply:
Maybe only split the ServiceGraphDataActionKeys in its own file? 

I feel other way to break this dependency could be to refactor the way utils/Authentication works by using `axios.interceptor` like here:  https://github.com/kiali/kiali-ui/blob/master/src/app/App.tsx#L24
 instead of passing the token to every method.

Reply:
Ok, I reverted the refactor. Now it only changes what is needed to fix the cycles.